buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java

```diff
@@ -323,9 +319,11 @@ private boolean isAnnotated(Class<?> clazz, Method method, Class<?> annotation)
     }
 
     private FileCollection getTestsClassPath() {
-        // This is doesn't need to be annotated with @Classpath because we only really care about the test source set
+        // Loading the classes does depend on the @Classpath but we can have better caching if we don't annotate this
```

Comment:
nit: What is the `@` for with classpath? It makes it look like an annotation, but I don't think it is?